packages/manifold/src/utils.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  chainIdToViemChain,
+  type FilterOperator,
+} from '@rabbitholegg/questdk-plugin-utils'
+import {
+  ERC1155_CONTRACT,
+  INSTANCEID_ABI,
+  ABI_MINT,
+  ABI_MULTI,
+} from './constants'
+import {
+  createPublicClient,
+  http,
+  type Address,
+  type ContractFunctionReturnType,
+} from 'viem'
+
+export interface ManifoldInput {
+  $abi: typeof ABI_MULTI | typeof ABI_MINT
+  creatorContractAddress: string
+  instanceId?: number | string
+  mintCount?: FilterOperator | undefined
+  mintFor?: string
+}
+
+export const shouldIncludeAbiMint = (
+  amount: FilterOperator | undefined,
+): boolean => {
+  if (amount == null) return true
+  if (typeof amount === 'object') {
+    if ('$gte' in amount && (amount.$gte as bigint) >= 2) return false
+    if ('$gt' in amount && (amount.$gt as bigint) >= 1) return false
+    if ('$eq' in amount && (amount.$eq as bigint) >= 2) return false
+    if ('$lt' in amount && (amount.$lt as bigint) <= 1) return false
+  } else {
+    return Number(amount) === 1
+  }
+  return true
+}
+
+const instanceIdCache: Record<string, number | undefined> = {}
```

Comment:
This cache won't persist - it's ephemeral since the SDK isn't a live service. Might as well just remove the cache. We would need to cache at the Backend layer since that's the layer that persists.